src/Tribe/Widget/Widget_Abstract.php

```diff
@@ -0,0 +1,416 @@
+		 * @param static $instance               Which instance of widget we are dealing with.
+		 */
+		$validate_arguments_map = apply_filters( "tribe__widget_{$registration_slug}_validate_arguments_map", $validate_arguments_map, $this );
+
+		return $validate_arguments_map;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_arguments() {
+
+		return $this->filter_arguments( $this->arguments );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function filter_arguments( $arguments ) {
+		/**
+		 * Applies a filter to instance arguments.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $arguments Current set of arguments.
+		 * @param static $instance  Which instance of widget we are dealing with.
+		 */
+		$arguments = apply_filters( 'tribe_widget_arguments', $arguments, $this );
+
+		$registration_slug = $this->get_registration_slug();
+
+		/**
+		 * Applies a filter to instance arguments based on the registration slug of the widget.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $arguments Current set of arguments.
+		 * @param static $instance  Which instance of widget we are dealing with.
+		 */
+		$arguments = apply_filters( "tribe_widget_{$registration_slug}_arguments", $arguments, $this );
+
+		return $arguments;
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_argument( $index, $default = null ) {
+		$arguments = $this->get_arguments();
+		$argument  = Arr::get( $arguments, $index, $default );
+
+		return $this->filter_argument( $argument, $index, $default );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function filter_argument( $argument, $index, $default = null  ) {
+		/**
+		 * Applies a filter to a specific widget argument, catch all for all widgets.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed  $argument The argument.
+		 * @param array  $index    Which index we indent to fetch from the arguments.
+		 * @param array  $default  Default value if it doesn't exist.
+		 * @param static $instance Which instance of widget we are dealing with.
+		 */
+		$argument = apply_filters( 'tribe_widget_argument', $argument, $index, $default, $this );
+
+		$registration_slug = $this->get_registration_slug();
+
+		/**
+		 * Applies a filter to a specific widget argument, to a particular registration slug.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed  $argument The argument value.
+		 * @param array  $index    Which index we indent to fetch from the arguments.
+		 * @param array  $default  Default value if it doesn't exist.
+		 * @param static $instance Which instance of widget we are dealing with.
+		 */
+		$argument = apply_filters( "tribe_widget_{$registration_slug}_argument", $argument, $index, $default, $this );
+
+		return $argument;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_default_arguments() {
+
+		return $this->filter_default_arguments( $this->default_arguments );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function filter_default_arguments( $default_arguments ) {
```

Comment:
```suggestion
	public function filter_default_arguments( array $default_arguments = [] ) {
```

Since we know and expect an array, let's type-hint it.